common-pg/src/main/scala/com/socrata/pg/store/PGSecondaryDatasetMapReader.scala

```diff
@@ -1,53 +1,167 @@
 package com.socrata.pg.store
 
-import java.sql.Connection
 import com.rojoma.simplearm.v2._
-import com.socrata.datacoordinator.id.{DatasetId, RollupName}
+import com.socrata.datacoordinator.id.{CopyId, DatasetId, RollupName}
+import com.socrata.datacoordinator.truth.metadata.sql.{BasePostgresDatasetMapReader, PostgresDatasetMapReader}
+import com.socrata.datacoordinator.truth.metadata.{CopyInfo, DatasetInfo, LifecycleStage, TypeNamespace}
 import com.socrata.datacoordinator.util.TimingReport
-import com.socrata.datacoordinator.truth.metadata.sql.{PostgresDatasetMapReader, BasePostgresDatasetMapReader}
-import com.socrata.datacoordinator.truth.metadata.{TypeNamespace, CopyInfo}
-import com.socrata.soql.types.SoQLType
+import org.joda.time.DateTime
+
+import java.sql.Connection
+import scala.reflect.ClassTag
 
-trait LocalRollupReaderOverride[CT] { self: BasePostgresDatasetMapReader[CT] =>
-  def localRollupsQuery = "SELECT name, soql, table_name FROM rollup_map WHERE copy_system_id = ?"
+trait LocalRollupReaderOverride[CT] {
+  self: BasePostgresDatasetMapReader[CT] =>
   override def rollups(copyInfo: CopyInfo): Seq[LocalRollupInfo] = {
     using(conn.prepareStatement(localRollupsQuery)) { stmt =>
       stmt.setLong(1, copyInfo.systemId.underlying)
       using(t("rollups", "copy_id" -> copyInfo.systemId)(stmt.executeQuery())) { rs =>
         val res = Vector.newBuilder[LocalRollupInfo]
```

Comment:
All of this mutable code can be made much nicer with something like 
```
  private def asIteratorOf[T](stmt: PreparedStatement)(f: ResultSet => T): Iterator[T] = {
    val resultSet = stmt.executeQuery()

    new Iterator[T] {
      def hasNext = resultSet.next()
      def next() = f(resultSet)
    }
  }

  def asVectorOf[T](stmt: PreparedStatement)(f: ResultSet => T): Vector[T] = asIteratorOf(stmt)(f).toVector
  ```

Reply:
Not my method, and did not change this much.
I have previously pitched using a syntax like:
`Iterator.continually(rs.next())
  .takeWhile(identity)
  .map(_ => Model(
      id = rs.getInt("id"),
      text = rs.getString("text")
   ))`
   
   But this was turned down in other PRs in favor of simplicity.

Reply:
that wouldn't quite work as takeWhile is assuming you're only emitting booleans.

Reply:
resultset.next is a boolean, and identity function just echoes that back as the continue condition